src/services/sync/strategies/consumerActions/UserAction.js

```diff
@@ -51,6 +50,51 @@ class UserAction extends BaseConsumerAction {
 		}
 	}
 
+	async autoMatchImportUser(schoolId, userUpdateObject) {
+		if (!isNotEmptyString(userUpdateObject.firstName, true) || !isNotEmptyString(userUpdateObject.lastName, true)) {
+			return;
+		}
+		const matchingLocalUsers = await UserRepo.findBySchoolAndName(
+			schoolId,
+			userUpdateObject.firstName,
+			userUpdateObject.lastName
+		);
+		if (!matchingLocalUsers || matchingLocalUsers.length !== 1) {
+			return;
+		}
+		const userMatch = matchingLocalUsers[0];
+		const foundImportUsers = await UserRepo.findImportUsersBySchoolAndName(
+			schoolId,
+			userUpdateObject.firstName,
+			userUpdateObject.lastName
+		);
+		if (foundImportUsers.length === 0) {
+			userUpdateObject.match = {
+				userId: userMatch._id,
+				matchedBy: 'auto',
+			};
+		} else {
+			// revoke other previously matched import users
+			await Promise.all(
+				foundImportUsers.map((foundImportUser) => {
+					delete foundImportUser.match;
```

Comment:
we only should delete auto matches here

Reply:
For now, we don't want to support reruns. We can discuss this later.